src/lib/support/CommonPersistentData.h

```diff
@@ -0,0 +1,83 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Contains a class handling creation of linked list of stored persistent data.
+ */
+
+#include <lib/support/PersistentData.h>
+
+namespace chip {
+
+typedef uint8_t entryIndex;
+constexpr entryIndex kUndefinedEntryIndex = 0;
+
+struct StoredDataList : public PersistentData<kPersistentBufferMax>
```

Comment:
Need doc comments on what the intent of this class is, what it stores and why.

Why is the entry_count and first_entry a uint_8? Are we assuming all possible chip lists will only need 8 bits?

Reply:
As stated in my previous comment, this is due to the legacy of the struct, initially being meant for a list of Fabrics. Size increase depend on future intended usage and could be modified in the event someone wants to implement a greater list of stored data. uint_8 is sufficient for the currently intended purpose of this class.

Reply:
The generic implementation now allows the user to specify a type of index when using the struct.